src/DeepBSDE.jl

```diff
@@ -151,16 +151,13 @@ function DiffEqBase.solve(
     prob = SDEProblem{false}(F, G, [x0;0f0], tspan, p3, noise_rate_prototype=noise)
 
     function neural_sde(init_cond)
-        map(1:trajectories) do j #TODO add Ensemble Simulation
-            predict_ans = Array(solve(prob, sdealg;
-                                         dt = dt,
-                                         u0 = init_cond,
-                                         p = p3,
-                                         save_everystep=false,
-                                         sensealg=DiffEqSensitivity.TrackerAdjoint(),
-                                         kwargs...))[:,end]
-            (X,u) = (predict_ans[1:(end-1)], predict_ans[end])
+        output_func(sol,i) = ((sol[end][1:end-1], sol[end][end]),false)
+        function prob_func(prob,i,repeat)
+            SDEProblem(prob.f , prob.g , init_cond, prob.tspan , prob.p ,noise_rate_prototype = prob.noise_rate_prototype)
```

Comment:
```suggestion
            SDEProblem(prob.f , prob.g , init_cond, prob.tspan , prob.p ,noise_rate_prototype = copy(prob.noise_rate_prototype))
```

Make sure you aren't reusing caches amongst threads. Best would be to just create `noise` as a vector matching `nthreads` so you don't collide, but given the cost of a solve here I think it's fine just to `copy`.